tee-worker/litentry/core/data-providers/src/graphql.rs

```diff
@@ -25,18 +25,25 @@ use itc_rest_client::{
 	rest_client::RestClient,
 	RestGet, RestPath,
 };
-use litentry_primitives::{EvmNetwork, IndexingNetwork, SubstrateNetwork};
-use log::*;
+use litentry_primitives::{EvmNetwork, SubstrateNetwork, SupportedNetwork};
+use log::debug;
 use serde::{Deserialize, Serialize};
 use std::{
 	collections::HashMap,
-	default::Default,
 	format, str,
 	string::{String, ToString},
 	vec,
 	vec::Vec,
 };
 
+pub trait TDFQuery<Query: ToGraphQL> {
+	fn verified_credentials_is_hodler(&mut self, params: Query) -> Result<IsHodlerOut, Error>;
+	fn verified_credentials_total_transactions(
+		&mut self,
+		params: Query,
+	) -> Result<Vec<TotalTxsStruct>, Error>;
+}
+
```

Comment:
maybe TDFxxx is not a good name. I would suggest try something more general, e.g. DataProviderQuery, etc.

Reply:
Yes, TDFxxx is inappropriate

Reply:
It depends though --- is this `___Query` type appropriate for _every_ data provider? Not just for the one we have now but for every provider we could use in the foreseeable future? If it's specifically tailored for TDF then I'd argue the more specific name should apply (although in that case it'd probably be more accurate to use `AchainableQuery` since that's the name of the service) :thinking: